specs/CustomDownload.md

```diff
@@ -0,0 +1,775 @@
+# Background
+<!-- TEMPLATE
+    Use this section to provide background context for the new API(s)
+    in this spec.
+
+    This section and the appendix are the only sections that likely
+    do not get copied into any official documentation, they're just an aid
+    to reading this spec.
+
+    If you're modifying an existing API, included a link here to the
+    existing page(s) or spec documentation.
+
+    For example, this section is a place to explain why you're adding this
+    API rather than modifying an existing API.
+
+    For example, this is a place to provide a brief explanation of some dependent
+    area, just explanation enough to understand this new API, rather than telling
+    the reader "go read 100 pages of background information posted at ...".
+-->
+We are redesigning the download experience by providing a DownloadStarting event, and exposing a Download object with all associated download metadata. With this you will be able to block downloads, save to a different path, and have access to the required metadata to build your own download UI.
+
+
+In this document we describe the updated API. We'd appreciate your feedback.
+
+# Description
+
+There are 3 parts to this API.
+
+1. DownloadStarting [Event] - used to intercept a download.
+    - Download (link to the download object below)
+    - Cancel
+    - Result file path (set)
+2. Download [Object]: This will give you all the metadata that you might need to block a download, and build UI.
+    - Uri
+    - Mime type
+    - Content disposition
+    - Download size in bytes
+    - Result file path (get)
+    - Progress size in bytes
+    - Estimated end time
+    - State (in progress, completed, interrupted)
+    - Pause/resume/cancel
+    - Deferral
```

Comment:
Deferral is on the event args not the download class, correct?

Reply:
Correct

Reply:
Great! Please move to event